oci/archive/oci_transport.go

```diff
@@ -0,0 +1,179 @@
+type ociArchiveReference struct {
+	resolvedFile  string
+	tag           string
+	tempDirectory string
+	ociRef        types.ImageReference
+}
+
+func (t ociArchiveTransport) Name() string {
+	return "oci-archive"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix
+// into an ImageReference.
+func (t ociArchiveTransport) ParseReference(reference string) (types.ImageReference, error) {
+	return ParseReference(reference)
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+func (t ociArchiveTransport) ValidatePolicyConfigurationScope(scope string) error {
+	return ocilayout.Transport.ValidatePolicyConfigurationScope(scope)
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an OCI ImageReference.
+func ParseReference(reference string) (types.ImageReference, error) {
+	var file, tag string
+	sep := strings.LastIndex(reference, ":")
+	if sep == -1 {
+		file = reference
+		tag = "latest"
+	} else {
+		file = reference[:sep]
+		tag = reference[sep+1:]
+	}
+	return NewReference(file, tag)
+}
+
+var refRegexp = regexp.MustCompile(`^([A-Za-z0-9._-]+)+$`)
+
+// NewReference returns an OCI reference for a file and a tag.
+func NewReference(file, tag string) (types.ImageReference, error) {
+	resolved, err := explicitfilepath.ResolvePathToFullyExplicit(file)
+	if err != nil {
+		return nil, err
+	}
+	// This is necessary to prevent file paths returned by PolicyConfigurationNamespaces
+	// from being ambiguous with values of PolicyConfigurationIdentity.
+	if strings.Contains(resolved, ":") {
+		return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", file, tag, resolved)
+	}
+	if !refRegexp.MatchString(tag) {
+		return nil, errors.Errorf("Invalid tag %s", tag)
+	}
+
+	return ociArchiveReference{resolvedFile: resolved, tag: tag}, nil
+}
+
+// createOciRef creates the oci reference of the image
+func createOciRef(ref *ociArchiveReference) error {
+	// dir is a temporary directory used to generate the OCI image
+	dir, err := ioutil.TempDir("/var/tmp", "oci")
+	if err != nil {
+		return errors.Errorf("error creating file %v", err)
+	}
+	ociRef, err := ocilayout.NewReference(dir, ref.tag)
+	if err != nil {
+		return err
+	}
+	ref.ociRef = ociRef
+	ref.tempDirectory = dir
+	return nil
+}
+
+func (ref ociArchiveReference) Transport() types.ImageTransport {
+	return Transport
+}
+
+// StringWithinTransport returns a string representation of the reference, which MUST be such that
+// reference.Transport().ParseReference(reference.StringWithinTransport()) returns an equivalent reference.
+func (ref ociArchiveReference) StringWithinTransport() string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.StringWithinTransport()
```

Comment:
This would return a path to the temporary directory, not to the user-specified archive. And like `ParseReference`, `StringWithinTransport` should not be modifying the filesystem.

 Again, copy&paste or extract into an independent helper.